src/graphics/text.c

```diff
@@ -84,6 +86,40 @@ int text_get_width(const uint8_t *str, font_t font)
     return width;
 }
 
+void text_fit_to_width(char *str, font_t font, int requested_width, const char *end_replacement)
```

Comment:
Maybe rename this to "text_ellipsize" and skip the `end_replacement` parameter?

Reply:
Good idea, that will also improve performance by only having to check the elipsis' width once per font type.